### ISSUE TITLE
Device is connected after 20 minutes on latest Windows 10 "Creators Update"

### ISSUE THREAD

**Issue by szszszsz** [Nitrokey/nitrokey-storage-firmware#33]
Windows 10 Creators Update
Details: https://support.nitrokey.com/t/nitrokey-storage-extrem-slow-after-windows-10-creators-update/431/4
Quote:
> After upgrading my computer to Windows 10 Creators Update (version 1703) mounting Nitrokey Storage is extremly slow.
> It tooks ~15-20 minutes after insserting the stick to recognise the unencrypted volume and also to recognise the stick in the nitro key app.
> After this time i can unlock the encrypted volume....and have to wait again to mount the unlocked volume.

**Comment by silvioboehme** [comment]
I checked this again on a Windows 10 Anniversary (1607) and stick is working fine.
Also checked this on a complete different machine with Win10 Creators Update (1703) with same issue as above.
So I think it's easy reproducible!

Just put a NitroKey Storage in a computer with a current Win 10.

**Comment by szszszsz** [comment]
I have retested this issue by the occasion of other check. Details:
Storage v0.45
Windows W10 1703 15063.338 - device connection is detected by the OS but it is not powering up the device.
Here is a Windows error message from Event Viewer:
```
Device USB\VID_20A0&PID_4109\0000000000000 had a problem starting.

Driver Name: usb.inf
Class Guid: {36fc9e60-c465-11cf-8056-444553540000}
Service: usbccgp
Lower Filters: 
Upper Filters: 
Problem: 0xA
Problem Status: 0xC000000D
```

**Comment by silvioboehme** [comment]
I don't see such a error message.
Does it appear directly after you plug in the stick or after "some" time?

**Comment by szszszsz** [comment]
This one is displayed in system's log. You can find it in `Device Manager` -> device's `Properties` -> `Event Log`

**Comment by silvioboehme** [comment]
On which device? Smartcard Reader?

**Comment by szszszsz** [comment]
![obraz](https://cloud.githubusercontent.com/assets/17005426/26589156/929e7c40-4557-11e7-9a74-dab0388f50a4.png)
In my case it is `USB Composite Device`.

**Comment by silvioboehme** [comment]
Nope...nothing. Everything is fine....except Nitrokey isn't working.

**Comment by szszszsz** [comment]
Just to note, what build exactly do you have? You can find it in `System` window.
One more idea: have you tried to enable `Legacy USB` (or similar) in BIOS/UEFI?

**Comment by silvioboehme** [comment]
1703 15063.296

**Comment by silvioboehme** [comment]
Windows W10 1703 15063.338 doesn't exists???

**Comment by szszszsz** [comment]
Indeed - typo, I meant .332, sorry.

**Comment by silvioboehme** [comment]
Btw..this is the first reaction after Microsoft relased the Creators Update!
... 2 month ago
... which is sad!

Also the bug title is a little bit "missleading". It is just unusable on a current windows 10!
After the stick is connected after about 10-20 minutes (connected means visible as a drive) it still isn't usable.
The drive letter appears twice, and the Nitro APP doesn't find the stick. (see another forum post!)
Also...if i copy a file to the unencrypted drive (on one drive letter), the file doesn't appear on the "other" drive....which initialy showed the same files.
 
Sadly...I have payed 99€ for nothing!

**Comment by juliomaranhao** [comment]
I have also this issue. Summarising my case: 

- UEFI only (Legacy Mode disabled) and Secure Boot disabled;
- Win 10 Pro, version 1703, build 15063.332, AKA "Creators Update", AKA "Current Win 10", AKA "I stayed with Anniversary as long as I could";
- Windows is in "Test Mode" to allow me to install unsigned drivers (VirtualSmartCard). Just for your info;
- Nitrokey Storage 8 GB, almost clean (never used), firmware version 0.45, App version 1.1;
- From image below: 2 virtual smartcard readers, Nitrokey and a gemalto (USB) smartcard reader. The command `opensc-tool -l` shows 3 smart cards but not Nitrokey. NitroApp can detect the Nitrokey.
![image](https://user-images.githubusercontent.com/791426/27004798-685d1c88-4de6-11e7-9cc3-720ea684ca88.png)
```
C:\Program Files\OpenSC Project\OpenSC\tools>opensc-tool.exe -l
# Detected readers (pcsc)
Nr.  Card  Features  Name
0    No              Gemalto USB Smart Card Reader 0
1    Yes             MySmartLogon Virtual Smart Card Reader 0
2    Yes             Virtual Smart Card Architecture Virtual PCD 0
```
![image](https://user-images.githubusercontent.com/791426/27004971-608bde8c-4dea-11e7-8645-3a9662c6ac3d.png)
"Não há drivers instalados para este dispositivo" = "No drivers installed for this device"
![image](https://user-images.githubusercontent.com/791426/27004969-4ad86f60-4dea-11e7-8b2e-12fc8609a3ff.png)

About the sequence:
- Nitrokey pluged;
- Device Manager lists "Nitrokey Storage" smartcard reader (above pics);
- Nitrokey App does not detect the Nitrokey and no drivers in windows explorer;
- Many minutes later:  2 drives appear (same label and contents) with 2 GB FAT32 each and NitroApp detects Nitrokey. Disc Manager also shows 2 new discs;
- `opensc-tool -l` still cannot show Nitrokey.
- And no errors in Event Viewer but I got this Information Event (translated) `Device USB\VID_20A0&PID_4109&MI_01\7&57a3ad1&0&0001 needs adicional installation.`
 
I have some time and knowledge to help with this bug ("windows advanced user") and I don't need Nitrokey now (I am evaluating it). But I agree with silvioboehme. The Nitrokey project needs help to reach a "product" status. I see this in the documentation. It's at least sparse. Maybe even scarse. And a Forum or FAQ is not the same as a Manual.

Anyway, I am here to help.

**Comment by juliomaranhao** [comment]
Me again. I decided to test my Nitrokey Storage 8GB in a Win7 Pro.

Results from 1st insertion:

![nitrokey win7 1st connection](https://user-images.githubusercontent.com/791426/27054313-eff4114a-4f96-11e7-891b-d67b02858a27.png)

Results from "remove and reinsert" AKA 2nd insertion:

![nitrokey win7 2nd connection](https://user-images.githubusercontent.com/791426/27054394-1dfe4894-4f97-11e7-9052-b6a641a04235.png)

Then, results from opensc-tool:
```
C:\Program Files\OpenSC Project\OpenSC\tools>opensc-tool -l
# Detected readers (pcsc)
Nr.  Card  Features  Name
0    Yes             Nitrokey Nitrokey Storage 0

C:\Program Files\OpenSC Project\OpenSC\tools>opensc-tool -n
Using reader with a card: Nitrokey Nitrokey Storage 0
CryptoStick v1.2 (OpenPGP v2.0)
```
Nitrokey App is working (version 1.1, firmware 0.45) and Explorer shows only 1 unencrypted drive (2 GB FAT32).

From this and my previous post:
- My Nitrokey Storage 8GB is NOT broken (works in Win7);
- In Win7 it takes 4 files (3 sys and 1 dll) in driver's list. All Microsoft files;
- In Win10 it takes only 1 file (1 sys) in driver's list. A Microsoft file;
- In Win10: "No drivers installed for this device";
- In Win7: "This device is working correctly" (Translated).

It's related to device detection and drivers installation. Even in Win7 I need to remove and reinsert for it to work (only 1st time ever, of course). This does not seem right as it's not a common behavior for other types of devices. Anyway it's a clue.

**Comment by juliomaranhao** [comment]
Two posts above I said my NK Storage was (almost) never used. I forgot I did a firmware upgrade from 0.4x (bought in december 2016) to 0.45. Now I only remember the PINs. I also tested the defaults. No luck with the firmware password.

So I am quitting as my NK is stuck with 0.45.

**Comment by jans23** [comment]
This issue should be solved by firmware [0.46](https://github.com/Nitrokey/nitrokey-storage-firmware/releases/tag/V0.46.0). Please [update](https://www.nitrokey.com/en/doc/firmware-update-storage) your firmware and try again.
